packages/itwinui-css/src/dialog/dialog.scss

```diff
@@ -3,144 +3,105 @@
 @import '../style/index';
 @import '../icon/index';
 
-@mixin iui-dialog-backdrop {
-  @include iui-reset;
-  z-index: 999;
-  isolation: isolate;
-  position: fixed;
+$iui-dialog-width: 380px;
+
+@mixin iui-dialog-wrapper {
+  position: relative;
+  overflow: hidden;
   top: 0;
   left: 0;
   width: 100%;
   height: 100%;
-  @include themed {
-    background-color: rgba(0, 0, 0, t(iui-opacity-4));
-  }
-
-  // Small/Resizable Dialog Animations for CSS
-  visibility: hidden;
-  opacity: 0;
-  transition: visibility $iui-speed-instant linear $iui-speed-fast, opacity $iui-speed-fast ease-out;
-
-  &.iui-dialog-visible {
-    visibility: visible;
-    opacity: 1;
-    // remove delay for entry animation so dialog is instantly visible
-    transition-delay: $iui-speed-instant;
-  }
-}
-
-@mixin iui-dialog-backdrop-relative {
-  position: relative;
-  overflow: hidden;
+  pointer-events: none;
+  transform: translateX(0); // creates containing block for position: fixed
 }
 
 @mixin iui-dialog {
+  z-index: 999;
+  isolation: isolate;
   border-radius: $iui-border-radius;
   box-shadow: $iui-elevation-24;
   box-sizing: border-box;
-  position: absolute;
+  position: fixed;
   padding: $iui-baseline $iui-m;
   box-sizing: border-box;
   padding: $iui-baseline $iui-m;
   overflow: hidden;
-  @include themed {
-    background-color: t(iui-color-background-1);
-  }
+  pointer-events: auto;
+  background-color: var(--iui-color-background-1);
   @media (forced-colors: active) {
     border: 1px solid;
   }
+
+  &:not(.iui-dialog-visible) {
+    visibility: hidden;
+    opacity: 0;
+  }
+
+  transition: visibility $iui-speed-instant linear, opacity $iui-speed-fast ease-out;
+  transition-delay: $iui-speed-fast, $iui-speed-instant;
+
+  &.iui-dialog-visible {
+    transition-delay: $iui-speed-instant; // remove delay for entry animation
+  }
 }
 
 @mixin iui-dialog-default {
-  .iui-dialog {
-    left: 50%;
-    top: 33%;
-    transform: translate(-50%, -33%);
-    max-width: 50%;
-    min-width: 380px;
-    max-height: 100vh;
-
-    @media screen and (max-width: 400px) {
-      max-width: 95%;
-      width: 95%;
-      min-width: 95%;
-    }
-  }
+  left: 50%;
+  top: 33%;
+  transform: translate(-50%, -33%);
+  max-width: Max(50%, $iui-dialog-width);
+  min-width: $iui-dialog-width;
+  max-height: 100vh;
```

Comment:
No `dvh` here?

Reply:
This says `max-height` which is kind of an edge case. I only added it for the `height` of full page dialog because otherwise it would _always_ overflow on iOS.